pyproject.toml

```diff
@@ -1,14 +1,197 @@
-[tool.black]
-extend-exclude = '''
-/(
-  | snapcraft_legacy
-  | tools
-  | docs/sphinx-resources
-)/
-'''
-# Targeting future versions as well so we don't have black reformatting code
-# en masse later.
-target-version = ["py312"]
+[project]
+name = "snapcraft"
+description = "Create snaps"
+dynamic = ["version", "readme"]
+dependencies = [
+    "attrs==24.2.0",
+    "catkin-pkg==1.0.0 ; sys_platform == 'linux'",
+    "click==8.1.7",
+    "craft-application~=4.4",
+    "craft-archives~=2.0",
+    "craft-cli>=2.15.0",
+    "craft-grammar>=2.0.1,<3.0.0",
+    "craft-parts==2.4.1",
+    "craft-platforms~=0.4",
+    "craft-store>=3.0.2,<4.0.0",
+    "cryptography==43.0.3", # Pinned due to https://github.com/canonical/snapcraft/issues/5217
+    "docutils<0.20", # Frozen until we can update sphinx dependencies.
+    "gnupg",
+    "jsonschema==2.5.1",
```

Comment:
This was pinned way back in 2021: https://github.com/canonical/snapcraft/commit/cbae1226406e09ace71d83fb534e174531ddba7f

We should probably update it.